glitter/tests/test_admin.py

```diff
@@ -131,6 +133,108 @@ def test_show_login(self):
         )
 
 
+@modify_settings(
+    INSTALLED_APPS={
+        'append': 'glitter.tests.sample',
+    },
+)
+class TestPermissions(TestCase):
+    def setUp(self):
+        # Permissions for objects we're testing
+        self.edit_page = Permission.objects.get_by_natural_key(
+            'edit_page', 'glitter_pages', 'page'
+        )
+        self.publish_page = Permission.objects.get_by_natural_key(
+            'publish_page', 'glitter_pages', 'page'
+        )
+        self.edit_book = Permission.objects.get_by_natural_key(
+            'edit_book', 'sample', 'book'
+        )
+        self.publish_book = Permission.objects.get_by_natural_key(
+            'publish_book', 'sample', 'book'
+        )
+
+        # Superuser
+        User = get_user_model()
+        self.superuser = User.objects.create_superuser(
+            username='superuser', email='', password=None
+        )
+
+        # Editor with editing permissions
+        self.editor = User.objects.create_user(username='editor', email='', password=None)
+        self.editor.is_staff = True
+        self.editor.save()
+        self.editor.user_permissions.add(self.edit_page, self.edit_book)
+
+        # Publisher with edit and publish permissions
+        self.publisher = User.objects.create_user(username='publisher', email='', password=None)
+        self.publisher.is_staff = True
+        self.publisher.save()
+        self.publisher.user_permissions.add(
+            self.edit_page, self.publish_page, self.edit_book, self.publish_book
+        )
+
+        # Staff with no editing permissions
+        self.staff = User.objects.create_user(username='staff', email='', password=None)
+        self.staff.is_staff = True
+        self.staff.save()
+
+        # Page with an unsaved page version
+        self.page = Page.objects.create(url='/test/', title='Test page')
+        self.page_version = Version.objects.create(
+            content_type=ContentType.objects.get_for_model(Page), object_id=self.page.id,
```

Comment:
Please follow https://docs.blanctools.com/standards/python-coding-standards.html#id9  For function calls which are very long or hard to read on a single line, then consider splitting one argument per line